### PR TITLE
Objects to represent VPC Lattice V2 request and response.

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*
n/a

*Description of changes:*

Objects to represent VPC Lattice V2 request and response as documented https://docs.aws.amazon.com/vpc-lattice/latest/ug/lambda-functions.html

*Target (OCI, Managed Runtime, both):* n/a

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by deki** [comment]
Great just created #443 for it (forgot to hit submit yesterday).

As per https://docs.aws.amazon.com/lambda/latest/dg/services-vpc-lattice.html#vpc-lattice-receiving-events `x509SanDns` should be added as well.

**Comment by msailes** [comment]
Added

File: aws-lambda-java-events/pom.xml [hunk 1]
```diff
@@ -50,6 +50,11 @@
             <artifactId>joda-time</artifactId>
             <version>2.10.8</version>
         </dependency>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
```
Comment (deki) [inline aws-lambda-java-events/pom.xml:55]: we shouldn't add that dependency for everyone

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/VpcLatticeV2RequestEvent.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        private Identity identity;
+        private String region;
+        /**
+         * Number of microseconds from the epoch
+         */
+        private String timeEpoch;
+
+        public LocalDateTime getlocalDateTime() {
+            // Truncating to milliseconds. Java 8 only supports Epoch with precision of milliseconds. Microseconds are supported in Java 9
+            long epochMilli = Long.parseLong(timeEpoch.substring(0, timeEpoch.length() - 3));
```
Comment (deki) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/VpcLatticeV2RequestEvent.java:56]: I'd add a null check even if the field is supposed to be always present. In addition provide a better method name for `getlocalDateTime`.

Comment (smirnoal) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/VpcLatticeV2RequestEvent.java:56]: if it is null, we'll get NPE in this line - and fail fast. I think this is expected behaviour in that case, as there's no reasonable fallback.

Comment (smirnoal) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/VpcLatticeV2RequestEvent.java:56]: On another note, I'd avoid truncating string in this line
my suggestion would be
```
long epochMicroseconds = Long.parseLong(timeEpoch);
long epochMilliseconds = epochMicroseconds / 1000;

return Instant.ofEpochMilli(epochMilliseconds).atZone(ZoneOffset.UTC).toLocalDateTime();
```

Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/VpcLatticeV2RequestEvent.java]: Done

File: aws-lambda-java-events/pom.xml (hunk 1 above)
Comment (msailes) [reply aws-lambda-java-events/pom.xml]: I've changed the dependency to the jetbrains one. There is no standard for this yet, which is a shame. I think it's important to describe the data of the events as best as we can. I hope that ```@Nullable``` will be an annotation that is added to more events in the future.

**Comment by sdelamo** [comment]
would it be possible to have Json samples of this event? it will allow us to test serialization.

**Pull request closed without merging**